### PR TITLE
Introducing 1.0.0

### PR DESCRIPTION
**Pull request by edwardcallahan**
Advances versioning to stable, 1.0.0 version.
Removes alpha and incubating labels.
Signed-off-by: Edward Callahan <edc@calncal.com>

### CODE REVIEW

File: deploy/olm-catalog/akka-cluster-operator/1.0.0/akka-cluster-operator-certified.v1.0.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+          }
+        }
+      ]
+spec:
+  apiservicedefinitions: {}
+  customresourcedefinitions:
+    owned:
+      - kind: AkkaCluster
+        name: akkaclusters.app.lightbend.com
+        version: v1alpha1
```
Comment (marcoderama) [inline deploy/olm-catalog/akka-cluster-operator/1.0.0/akka-cluster-operator-certified.v1.0.0.clusterserviceversion.yaml:76]: Any plans to bump this up to beta at least now?

File: deploy/olm-catalog/akka-cluster-operator/0.2.1/akka-cluster-operator-certified.v0.2.1.clusterserviceversion.yaml [hunk 2]
```diff
@@ -170,7 +170,7 @@ spec:
                       - name: OPERATOR_NAME
                         value: akka-cluster-operator
                       - name: RELATED_IMAGE_CONTROLLER
-                        value: registry.connect.redhat.com/lightbend/akkacluster-operator:v0.2.0   
+                        value: registry.connect.redhat.com/lightbend/akka-cluster-operator-certified:0.2.0
```
Comment (chbatey) [inline deploy/olm-catalog/akka-cluster-operator/0.2.1/akka-cluster-operator-certified.v0.2.1.clusterserviceversion.yaml:173]: should this be version 0.2.1?

File: deploy/olm-catalog/akka-cluster-operator/0.2.2/akka-cluster-operator-certified.v0.2.2.clusterserviceversion.yaml [hunk 3]
```diff
@@ -170,7 +170,7 @@ spec:
                       - name: OPERATOR_NAME
                         value: akka-cluster-operator
                       - name: RELATED_IMAGE_CONTROLLER
-                        value: registry.connect.redhat.com/lightbend/akkacluster-operator:v0.2.0   
+                        value: registry.connect.redhat.com/lightbend/akka-cluster-operator-certified:0.2.0
```
Comment (chbatey) [inline deploy/olm-catalog/akka-cluster-operator/0.2.2/akka-cluster-operator-certified.v0.2.2.clusterserviceversion.yaml:173]: And this 0.2.2?

File: deploy/olm-catalog/akka-cluster-operator/0.2.3/akka-cluster-operator-certified.v0.2.3.clusterserviceversion.yaml [hunk 4]
```diff
@@ -170,7 +170,7 @@ spec:
                       - name: OPERATOR_NAME
                         value: akka-cluster-operator
                       - name: RELATED_IMAGE_CONTROLLER
-                        value: registry.connect.redhat.com/lightbend/akkacluster-operator:v0.2.2   
+                        value: registry.connect.redhat.com/lightbend/akka-cluster-operator-certified:0.2.2
```
Comment (chbatey) [inline deploy/olm-catalog/akka-cluster-operator/0.2.3/akka-cluster-operator-certified.v0.2.3.clusterserviceversion.yaml:173]: And this 0.2.3?

File: deploy/olm-catalog/akka-cluster-operator/1.0.0/akka-cluster-operator-certified.v1.0.0.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,327 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: akka-cluster-operator-certified.v1.0.0
+  namespace: placeholder
+  annotations:
+    capabilities: Seamless Upgrades
+    categories: Application Runtime
+    certified: 'true'
+    containerImage: 'registry.connect.redhat.com/lightbend/akka-cluster-operator-certified:0.2.2'
```
Comment (chbatey) [inline deploy/olm-catalog/akka-cluster-operator/1.0.0/akka-cluster-operator-certified.v1.0.0.clusterserviceversion.yaml:10]: And this 1.0.0?

File: deploy/olm-catalog/akka-cluster-operator/1.0.0/akka-cluster-operator.v1.0.0.clusterserviceversion.yaml [hunk 6]
```diff
@@ -0,0 +1,320 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: akka-cluster-operator.v0.2.3
+  namespace: placeholder
+  annotations:
+    capabilities: Seamless Upgrades
+    categories: Application Runtime
+    certified: 'false'
+    containerImage: 'lightbend-docker-registry.bintray.io/lightbend/akkacluster-operator:v0.2.2'
```
Comment (chbatey) [inline deploy/olm-catalog/akka-cluster-operator/1.0.0/akka-cluster-operator.v1.0.0.clusterserviceversion.yaml:10]: version 1.0.0?

**Review by chbatey — COMMENTED** [review]
I think the versions are wrong but i also am not sure how all this works so I could be wrong!

File: deploy/olm-catalog/akka-cluster-operator/1.0.0/akka-cluster-operator-certified.v1.0.0.clusterserviceversion.yaml (hunk 1 above)
Comment (chbatey) [reply deploy/olm-catalog/akka-cluster-operator/1.0.0/akka-cluster-operator-certified.v1.0.0.clusterserviceversion.yaml:76]: @edwardcallahan when you said removing alpha labels were you meaning changing this? Or something else?

Comment (edwardcallahan) [reply deploy/olm-catalog/akka-cluster-operator/1.0.0/akka-cluster-operator-certified.v1.0.0.clusterserviceversion.yaml:76]: Good question. I specifically left the protocol version at `v1apha1` as to not commit us to more than alpha. Yes, this should move out of alpha. I only changed the release version to 1.0.0 and removed Alpha from the labels on the Marketplace. Again, seeking to restrain commitments to minimal needed for marketplace and avoiding binding us to the protocol definition.

File: deploy/olm-catalog/akka-cluster-operator/0.2.1/akka-cluster-operator-certified.v0.2.1.clusterserviceversion.yaml (hunk 2 above)
Comment (edwardcallahan) [reply deploy/olm-catalog/akka-cluster-operator/0.2.1/akka-cluster-operator-certified.v0.2.1.clusterserviceversion.yaml:173]: Actually, no. If you look at the [images in the project](https://connect.redhat.com/project/2073491/view), you'll see a 0.2.0, 0.2.2 and 1.0.1. Of course you could publish an 0.2.0 tag but there is not currently such a tag in the certified registry.

File: deploy/olm-catalog/akka-cluster-operator/0.2.2/akka-cluster-operator-certified.v0.2.2.clusterserviceversion.yaml (hunk 3 above)
Comment (edwardcallahan) [reply deploy/olm-catalog/akka-cluster-operator/0.2.2/akka-cluster-operator-certified.v0.2.2.clusterserviceversion.yaml:173]: Yes.

File: deploy/olm-catalog/akka-cluster-operator/1.0.0/akka-cluster-operator-certified.v1.0.0.clusterserviceversion.yaml (hunk 5 above)
Comment (edwardcallahan) [reply deploy/olm-catalog/akka-cluster-operator/1.0.0/akka-cluster-operator-certified.v1.0.0.clusterserviceversion.yaml:10]: Yes, if there was a 1.0.1 image, but it got eaten by the verification tooling issue and I ended up w/ 1.0.1 as my tag. Could push a 1.0.0 tag image to keep things clean. Can also run a metadata only 1.0.1 release and immediately follow up w/ a 1.0.2 to get things tidy again.

File: deploy/olm-catalog/akka-cluster-operator/0.2.3/akka-cluster-operator-certified.v0.2.3.clusterserviceversion.yaml (hunk 4 above)
Comment (edwardcallahan) [reply deploy/olm-catalog/akka-cluster-operator/0.2.3/akka-cluster-operator-certified.v0.2.3.clusterserviceversion.yaml:173]: This was likely a metadata only release. I recall the driver of the change being the metadata (yaml), so I pushed a new CSV w/out a matching 0.2.3 container. Sub-optimal, I admit.

File: deploy/olm-catalog/akka-cluster-operator/1.0.0/akka-cluster-operator.v1.0.0.clusterserviceversion.yaml (hunk 6 above)
Comment (edwardcallahan) [reply deploy/olm-catalog/akka-cluster-operator/1.0.0/akka-cluster-operator.v1.0.0.clusterserviceversion.yaml:10]: Yes, if you push a 1.0.0 tag to [bintray](https://bintray.com/lightbend/registry/lightbend%3Aakkacluster-operator).

**Comment by chbatey** [comment]
Thanks for taking a look Ed. I'll make the version changes and push 1.0 to bintray.

**Comment by chbatey** [comment]
Closing in favor of https://github.com/lightbend/akka-cluster-operator/pull/43

**Pull request closed without merging**